witness/src/Dscp/Witness/Logic/Getters.hs

```diff
@@ -138,20 +134,27 @@ getAccountTxs address = runSdMRead $ loadTxs >>= mapM getTx
 ----------------------------------------------------------------------------
 
 -- | Safely get transaction.
-getTxMaybe :: HasWitnessConfig => GTxId -> SdM (Maybe GTxInBlock)
+getTxMaybe :: HasWitnessConfig => GTxId -> SdM (Maybe (WithBlock GTxWitnessed))
 getTxMaybe gTxId = do
     SD.queryOne gTxId >>= \case
         Nothing -> pure Nothing
         Just TxBlockRef{..} ->
             getBlockMaybe tbrBlockRef >>= pure . \case
                 Nothing -> Nothing
-                Just block -> fmap (GTxInBlock $ Just block) . (^? ix tbrTxIdx) . bbTxs . bBody $ block
+                Just block -> fmap (WithBlock $ Just block) . (^? ix tbrTxIdx) . bbTxs . bBody $ block
 
 -- | Resolves transaction, throws exception if it's absent.
-getTx :: HasWitnessConfig => GTxId -> SdM GTxInBlock
+getTx :: HasWitnessConfig => GTxId -> SdM (WithBlock GTxWitnessed)
 getTx gTxId = do
     tM <- getTxMaybe gTxId
     maybe (SD.throwLocalError $ LETxAbsent $
               "Can't get transaction with id " <> pretty gTxId)
           pure
           tM
+
+----------------------------------------------------------------------------
+-- Publication getters
+----------------------------------------------------------------------------
+
+getPrivateTipHash :: HasWitnessConfig => Address -> SdM_ chgacc (Maybe PrivateHeaderHash)
```

Comment:
Why `chgacc` is abstracted here?

Reply:
It allows running an action with both `runSdRead` and `runSdMempoolRead`.